packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -35,11 +35,8 @@ import { hasSome } from "../util/index.js";
  * (or is undefined if there was no net change):
  * * Append - when one or more commits are appended to the head of the branch, for example via
  * a change applied by the branch's editor, or as a result of merging another branch into this one
- * * Remove - when one or more commits are removed from the head of the branch. This occurs
- * when a transaction is aborted and all commits pending in that transaction are removed.
- * * Replace - when an operation simultaneously removes and appends commits. For example, when this
- * branch is rebased and some commits are removed and replaced with rebased versions, or when a
- * transaction completes and all pending commits are replaced with a single squash commit.
+ * * Remove - when one or more commits are removed from the head of the branch.
+ * * Rebase - when a rebase operation replaces commits at the head of the branch with rebased versions of those commits.
```

Comment:
This description sounds like the branch would go from `A<-B<-C<-D` to `A<-B<-C'<-D'`, but that leaves out the introduction of new commits from the branch we're rebasing onto (e.g., X and Y in `A<-B<-X<-Y<-C'<-D'`).

Reply:
I tweaked it to be correct, though I didn't spend too much time on making it incredibly prescriptive given that we want to kill it or replace it R.E. other comment.